python/social_bot/envs/grocery_ground.py

```diff
@@ -0,0 +1,232 @@
+
+        self._agent = self._world.get_agent()
+        self._pr2_joints = list(
+            filter(
+                lambda s: s.find('fl_caster_r_wheel_joint') != -1 or s.find('fr_caster_l_wheel_joint') != -1,
+                self._agent.get_joint_names()))
+        logger.info("joints to control: %s" % self._pr2_joints)
+        self._objects_names = [
+            'coke_can_0', 'coke_can_1', 'coke_can_2', 'coke_can_3',
+            'coke_can_4', 'table', 'grey_wall', 'cube_20k', 'caffe_table',
+            'bookshelf', 'car_wheel_0', 'car_wheel_1', 'trash_can',
+            'plastic_cup_0', 'plastic_cup_1', 'plastic_cup_2', 'plastic_cup_3',
+            'plastic_cup_4'
+        ]
+        self._goal_name = 'bookshelf'
+
+        self._teacher = teacher.Teacher(False)
+        task_group = teacher.TaskGroup()
+        task_group.add_task(GroceryGroundGoalTask())
+        self._teacher.add_task_group(task_group)
+
+        self._with_language = with_language
+        self._use_image_obs = use_image_obs
+        self._resized_image_size = (84, 84)
+
+        obs = self.reset()
+        if self._use_image_obs:
+            obs_data_space = gym.spaces.Box(
+                low=0, high=1, shape=obs.shape, dtype=np.uint8)
+        else:
+            obs_data_space = gym.spaces.Box(
+                low=-50, high=50, shape=obs.shape, dtype=np.float32)
+        control_space = gym.spaces.Box(
+            low=-20, high=20, shape=[len(self._pr2_joints)], dtype=np.float32)
+
+        if self._with_language:
+            self.observation_space = gym.spaces.Dict(
+                data=obs_data_space, sentence=DiscreteSequence(128, 24))
+            self._action_space = gym.spaces.Dict(
+                control=control_space, sentence=DiscreteSequence(128, 24))
+        else:
+            self.observation_space = obs_data_space
+            self.action_space = control_space
+
+    def reset(self):
+        """
+        Reset the environment
+        Args: None
+        Returns: Observaion of the first step
+        """
+        self._world.reset()
+        self._world.step(5)
+        self._collision_cnt = 0
+        self._cum_reward = 0.0
+        self._steps_in_this_episode = 0
+        self._goal = self._world.get_agent(self._goal_name)
+        self._teacher.reset(self._agent, self._world)
+        teacher_action = self._teacher.teach("")
+        if self._with_language:
+            obs_data = self._get_observation()
+            obs = OrderedDict(data=obs_data, sentence=teacher_action)
+        else:
+            obs = self._get_observation()
+        return obs
+
+    def _get_observation(self):
+        if self._use_image_obs:
+            obs_data = np.array(
+                self._agent.get_camera_observation(
+                    "default::pr2::pr2::head_tilt_link::head_mount_prosilica_link_sensor"
+                ),
+                copy=False)
+            obs_data = PIL.Image.fromarray(rgb2gray(obs_data)).resize(
```

Comment:
we should use color image. gray image loses information.

Reply:
done